test_aodndata/aatams/test_aatams_sattag.py

```diff
@@ -18,13 +18,15 @@
 TEST_ROOT = os.path.join(os.path.dirname(__file__))
 
 DM_GOOD_ZIP = os.path.join(TEST_ROOT, "simple_dm.zip")
-DM_SINGLE_CSV = os.path.join(TEST_ROOT, "metadata_dm.csv")
+DM_SINGLE_CSV = os.path.join(TEST_ROOT, "metadata_ct111_dm.csv")
 
 NRT_FIRST_ZIP = os.path.join(TEST_ROOT, "first_nrt_batch", "ct155_nrt.zip")
 NRT_SECOND_ZIP = os.path.join(TEST_ROOT, "second_nrt_batch", "ct155_nrt.zip")
 NRT_NEW_CAMPAIGN = os.path.join(TEST_ROOT, "new_ct156_nrt.zip")
-NRT_SINGLE_CSV = os.path.join(TEST_ROOT, "metadata_nrt.csv")
+NRT_SINGLE_CSV = os.path.join(TEST_ROOT, "metadata_ct111_nrt.csv")
 NRT_EMPTY_DIVE_ZIP = os.path.join(TEST_ROOT, "emptydive_ct999_nrt.zip")
+NRT_MIXED_CAMPAIGN = os.path.join(TEST_ROOT, "mixcampaign_ct555_nrt.zip")
+NRT_CSV_CAMPAIGN_MISMATCH = os.path.join(TEST_ROOT, "csvcampaignmismatch_ct999_nrt.zip")
```

Comment:
Just FYI, you can also create test zip files on the fly with [`aodncore.testlib.make_zip`](https://github.com/aodn/python-aodncore/blob/master/aodncore/testlib/testutil.py#L187)

Reply:
Yeah, I may refactor into csv-folders if we ever need more zips.

Reply:
I just quickly tried to move towards using makezip for all current zip files, but this would incur in way more code, more file creation,  and less re-use. The problem lies with `TemporaryDirectory` removing the temporary folder when the scope finishes. Hence, I would need to copy/paste the Temporary scope into all tests, build zips in every test, rebuild several zips in the tests that use more than one zip file, etc. Here they are a single line definition.

The proper solution would involve a TemporaryDirectory scope that is bounded to the test runtime - a thing pytest already got.


I'm leaving as is.